### ISSUE TITLE
Backlog: Normalize how optional form fields behave

### ISSUE THREAD

**Issue by kitajchuk** [jambonz/jambonz-webapp#123]
**TL;DR: Option 2 is the way to go**

**Checklist:**

- [x] Settings form (fine as is—done)
- [x] Account form (fine as is—done)
- [x] Application form (fine as is—done)
- [ ] Carrier form (needs to be converted to option 2 behavior)
- [ ] Speech form (needs to be converted to option 2 behavior—can work the same but is different because it doesn't use `checkzone`)

Currently we handle optional form fields a couple of ways currently across the app. Most notably there are some optional field sets on the Carrier form in which un-checking the box wipes them out of local state. However on the Account/Application forms we have optional HTTP basic auth for web hooks and un-checking the box there leaves the entered values in tact—e.g. they would post on save and persist on the backend and the `checkzone` would toggle open to reveal them again. The Settings form is like this last method however it saves the field value in temp state and empties the field and then replaces it since this `checkzone` isn't using the `hidden` prop. _Perhaps we need to decide on an official house rule about these optional fields and normalize them all across these implementations?_

If it's the case that we normalize to one style there are two viable options:

1. Un-checking the box empties the fields that are toggled by the box. The user then has to check the box again and enter them again if they decide they want to do that. This means values are not persisted for un-checked boxes in local state and are therefore deleted on the backend when saved—even if they existed on the backend.
2. Un-checking the box doesn't empty the fields and they are persisted in local state and saved to the backend. In this case for the Checkzones the act of saving ultimately toggles the zone back open because the fields are saved on the backend.

**Option 1 currently looks like this:**

https://user-images.githubusercontent.com/438711/192866785-9b515326-35a4-4d3d-8c89-90593de14061.mp4

**Option 2 currently looks like this:**

https://user-images.githubusercontent.com/438711/192866407-0565f8af-c16c-487e-8ba2-ca24e1790469.mp4

Personally I prefer option 2 because as we see in the case of the Carrier form I already have Outbound values saved on the backend and the simple act of toggling that field wipes them in local state. If I were to then save the form they would get deleted on the backend. Potentially this is not desirable TBH. So for this stuff I am leaning towards always go with what the user entered/did. If, as a user, I really don't want those fields anymore I can delete the values and save. But simply toggling the fields hidden or shown seems alarming to just automagically lose everything.
